pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousReader.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.spark;
+
+import com.google.cloud.pubsublite.Partition;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.proto.PartitionCursor;
+import com.google.cloud.pubsublite.proto.Subscription;
+import com.google.cloud.pubsublite.proto.TopicPartitions;
+import com.google.cloud.pubsublite.v1.AdminServiceClient;
+import com.google.cloud.pubsublite.v1.CursorServiceClient;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.Serializable;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.sources.v2.reader.InputPartition;
+import org.apache.spark.sql.sources.v2.reader.streaming.ContinuousReader;
+import org.apache.spark.sql.sources.v2.reader.streaming.Offset;
+import org.apache.spark.sql.sources.v2.reader.streaming.PartitionOffset;
+import org.apache.spark.sql.types.StructType;
+
+public class PslContinuousReader implements ContinuousReader, Serializable {
+
+  private final PslDataSourceOptions options;
+  private final SubscriptionPath subscriptionPath;
+  private final AdminServiceClient adminServiceClient;
```

Comment:
Good idea :)

Reply:
I now have ServiceClient and Client, ServiceClient is needed since I need to pass the creds.
We should actually modify XXXClientSettings to accept creds but could be in a later PR.

Reply:
> We should actually modify XXXClientSettings to accept creds

I think they accept a ServiceClient? So instead, in the options method, just construct the service client in a function body and immediately wrap it with our clients.

Reply:
They do, but my point is, the client level api should just accept creds, so in option method, I don't need to even get into the serviceclient level, and worry about for example I need to remember addDefaultSettings blahblah.